charm/kafka/lib/charms/layer/kafka.py

```diff
@@ -13,72 +13,89 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import ipaddress
 import os
-import re
 import shutil
+import re
 import socket
 
 from pathlib import Path
+from base64 import b64encode
 
 from charmhelpers.core import hookenv, host
 from charmhelpers.core.templating import render
 
+from charms.reactive.relations import RelationBase
+
+from charms.layer import snap
+
+KAFKA_PORT = 9093
+KAFKA_SNAP = 'kafka'
+KAFKA_SERVICE = 'snap.{}.kafka.service'.format(KAFKA_SNAP)
+BOOTSTRAP_ZK_SERVICE = 'snap.{}.zookeeper.service'.format(KAFKA_SNAP)
+KAFKA_SNAP_DATA = '/var/snap/{}/common'.format(KAFKA_SNAP)
+KAFKA_KEYTOOL_PATH = '/snap/{}/current/usr/lib/jvm/default-java/bin/keytool'.format(
+    KAFKA_SNAP
+)
 
-KAFKA_PORT=9093
-KAFKA_SNAP='kafka'
-KAFKA_SERVICE='snap.{}.kafka.service'.format(KAFKA_SNAP)
-KAFKA_SNAP_DATA='/var/snap/{}/common'.format(KAFKA_SNAP)
 
 class Kafka(object):
     def open_ports(self):
+        '''
+        Attempts to open the Kafka port.
+        '''
         hookenv.open_port(KAFKA_PORT)
 
     def close_ports(self):
+        '''
+        Attempts to close the Kafka port.
+        '''
         hookenv.close_port(KAFKA_PORT)
 
-    def configure_kafka(self, zk_units, log_dir=None):
-        # Get ip:port data from our connected zookeepers
+    def install(self, zk_units=[], log_dir='logs'):
+        '''
+        Generates client-ssl.properties and server.properties with the current
+        system state.
+        '''
         zks = []
-        for unit in zk_units:
+        for unit in zk_units or self.get_zks():
```

Comment:
Please revert. Kafka.install is always called from the reactive charm code below with the zookeeper relation data, so get_zks is unnecessary.